### DISCUSSION TITLE
`sea_orm_migration::schema::timestamp_null` Invalid default value in mysql

### DISCUSSION THREAD

**Discussion by laojianzi** [SeaQL/sea-orm#2253] [Q&A]
https://github.com/SeaQL/sea-orm/blob/987cb38aaf4ebe5554dcce3eba7b88e981563a87/sea-orm-migration/src/schema.rs#L320

`timestamp_null` looks like it's telling me to declare that NULL is allowed, whereas the SQL provided by sea-orm doesn't have NULL and doesn't have NOT NULL, and leaves it up to the default behavior of the database's type.

When I tested MySQL 5.7, the database returned an error:
```sql
`column_name` timestamp -- ERROR: Invalid default value for 'column_name'
```

---

Using `null` solves the problem, but I think most people assume that `timestamp_null` already includes this behavior
```rust
timestamp_null(Users::ColumnAt).null()
```

So when the semantics of `timestamp_null` is null, should we show that declaring the field is allowed to be NULL instead of leaving it to the default behavior of the database?
```sql
`column_name` timestamp NULL
```

**Comment by tyt2y3** [comment]
I agree with you. It would be better to add a `NULL` explicitly. Although is there a case where the default behavior is not `NULL`?

**Reply by laojianzi** [reply]
Yes. we can represent the default database behavior in other ways, such as `type_default()`, since the current `type()` will use `not_null()`

https://github.com/SeaQL/sea-orm/blob/987cb38aaf4ebe5554dcce3eba7b88e981563a87/sea-orm-migration/src/schema.rs#L316-L318

**Comment by billy1624** [comment]
Hey @laojianzi, thanks for the report! I believe this is a special case for MySQL 5.7, https://dba.stackexchange.com/a/314900.

However, it's always good to have a explicit `NULL` for nullable column. Please check the PR:
- https://github.com/SeaQL/sea-orm/pull/2255
